nemo_rl/evals/eval.py

```diff
@@ -116,30 +110,140 @@ def setup(
+        if ckpt_cfg_path.exists():
+            with open(ckpt_cfg_path, "r") as f:
+                _ckpt_master_cfg = yaml.safe_load(f)
+            policy_config = _ckpt_master_cfg.get("policy")
+
+        if policy_config is None:
+            raise ValueError(
+                "Policy configuration is required for refitting but was not provided in the evaluation YAML "
+                "and could not be found in the checkpoint (config.yaml)."
+            )
+
+        base_model_name = policy_config["model_name"]
+        tokenizer_path_for_vllm_and_hf = str(
+            dcp_tokenizer_path if dcp_tokenizer_path.exists() else base_model_name
+        )
+        trust_remote_code_tokenizer = policy_config.get("tokenizer", {}).get(
+            "trust_remote_code"
+        )
+
+        # 1. Initialize VllmGeneration with base model and load_format="dummy"
+        print(
+            f"  ▶ Initializing VllmGeneration with base model '{base_model_name}' (dummy load)..."
+        )
+        vllm_init_gen_cfg = generation_config.copy()
+        vllm_init_gen_cfg["model_name"] = base_model_name
+        # Ensure VllmGeneration uses the tokenizer from DCP if available
+        vllm_init_gen_cfg["tokenizer_name"] = tokenizer_path_for_vllm_and_hf
+        # Pass trust_remote_code for tokenizer (assuming GenerationConfig/VllmGeneration supports tokenizer_trust_remote_code)
+        vllm_init_gen_cfg["tokenizer_trust_remote_code"] = trust_remote_code_tokenizer
+        vllm_init_gen_cfg["trust_remote_code"] = policy_config.get("trust_remote_code")
+
+        current_vllm_opts = vllm_init_gen_cfg.get("vllm_cfg", {}).copy()
+        current_vllm_opts["load_format"] = "dummy"
+        vllm_init_gen_cfg["vllm_cfg"] = current_vllm_opts
+        vllm_model = VllmGeneration(cluster=cluster, config=vllm_init_gen_cfg)
+        print(f"    ✓ VllmGeneration initialized for base '{base_model_name}'.")
+
+        # 2. Initialize temporary HfPolicy to load DCP weights
+        print(
+            f"  ▶ Loading DCP weights into temporary HfPolicy from: {dcp_weights_path}..."
+        )
+        tokenizer_for_hf_policy = AutoTokenizer.from_pretrained(
+            tokenizer_path_for_vllm_and_hf,
+            trust_remote_code=trust_remote_code_tokenizer,
+        )
+
+        temp_hf_policy_config = policy_config.copy()
+        temp_hf_policy = HfPolicy(
+            cluster=cluster,
+            config=temp_hf_policy_config,
+            tokenizer=tokenizer_for_hf_policy,
+            weights_path=str(dcp_weights_path),
+            init_optimizer=False,
+        )
+        print(f"    ✓ Temporary HfPolicy loaded with weights from {dcp_weights_path}.")
+
+        # 3. Perform Refit
+        print("  ▶ Refitting weights from HfPolicy to VllmGeneration...")
+        refit_buffer_size = policy_config.get("refit_buffer_size_gb")
+        vllm_model.finish_generation()
+        refit_policy_generation(temp_hf_policy, vllm_model, refit_buffer_size)
```

Comment:
`refit_buffer_size_gb` param is removed at the latest main branch, `_refit_buffer_size_gb` is only for test purpose now.
```suggestion
        vllm_model.finish_generation()
        refit_policy_generation(temp_hf_policy, vllm_model)
```